src/ImageSharp/Advanced/ParallelRowIterator.Wrappers.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.Memory;
+
+namespace SixLabors.ImageSharp.Advanced
+{
+    /// <content>
+    /// Utility methods for batched processing of pixel row intervals.
+    /// Parallel execution is optimized for image processing based on values defined
+    /// <see cref="ParallelExecutionSettings"/> or <see cref="Configuration"/>.
+    /// Using this class is preferred over direct usage of <see cref="Parallel"/> utility methods.
+    /// </content>
+    public static partial class ParallelRowIterator
+    {
+        private readonly struct WrappingRowIntervalInfo
+        {
+            public readonly int MinY;
+            public readonly int MaxY;
+            public readonly int StepY;
+            public readonly int MaxX;
+
+            public WrappingRowIntervalInfo(int minY, int maxY, int stepY)
+                : this(minY, maxY, stepY, 0)
+            {
+            }
+
+            public WrappingRowIntervalInfo(int minY, int maxY, int stepY, int maxX)
+            {
+                this.MinY = minY;
+                this.MaxY = maxY;
+                this.StepY = stepY;
+                this.MaxX = maxX;
+            }
+        }
+
+        private readonly struct WrappingRowIntervalOperation<T>
```

Comment:
Shouldn't this be `RowIntervalOperationWrapper<T>` instead? Took me a for a while to figure out it is exactly the opposite as it's name :)

Reply:
Haha, yeah no problem.